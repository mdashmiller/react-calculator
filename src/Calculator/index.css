.store, .display {
	text-align: right;
	border-radius: 0px;
}

input {
	border-radius: 50px;
}
