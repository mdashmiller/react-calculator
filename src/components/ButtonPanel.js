import React from 'react'
import PropTypes from 'prop-types'

const ButtonPanel = ({ handleClick, handleKeyDown, calculate }) => 
	// sends values from button to App state
	<div className="wrapper" onKeyDown={handleKeyDown}>
		<div className="top-row">
			<button className="button func" onClick={ () => handleClick('ce') } autoFocus>ce</button>
			<button className="button func" onClick={ () => handleClick('c') }>c</button>
			<button className="button mod" onClick={ () => handleClick('+/-') }>+/-</button>
		</div>
		<div className="button-row">
			<button className="button number" onClick={ () => handleClick('7') }>7</button>
			<button className="button number" onClick={ () => handleClick('8') }>8</button>
			<button className="button number" onClick={ () => handleClick('9') }>9</button>
			<button className="button op" onClick={ () => handleClick('/') }>/</button>
		</div>
		<div className="button-row">
			<button className="button number" onClick={ () => handleClick('4') }>4</button>
			<button className="button number" onClick={ () => handleClick('5') }>5</button>
			<button className="button number" onClick={ () => handleClick('6') }>6</button>
			<button className="button op" onClick={ () => handleClick('x') }>x</button>
		</div>
		<div className="button-row">
			<button className="button number" onClick={ () => handleClick('1') }>1</button>
			<button className="button number" onClick={ () => handleClick('2') }>2</button>
			<button className="button number" onClick={ () => handleClick('3') }>3</button>
			<button className="button op" onClick={ () => handleClick('-') }>-</button>
		</div>
		<div className="button-row">
			<button className="button number" onClick={ () => handleClick('0') }>0</button>
			<button className="button mod" onClick={ () => handleClick('.') }>.</button>
			<button className="button func" id="calc" onClick={calculate}>=</button>
			<button className="button op" onClick={ () => handleClick('+') }>+</button>
		</div>
	</div>	

ButtonPanel.propTypes = {
	handleClick: PropTypes.func.isRequired,
	handleKeyDown: PropTypes.func.isRequired,
	calculate: PropTypes.func.isRequired
}

export default ButtonPanel
