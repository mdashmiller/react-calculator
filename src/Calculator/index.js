import React from 'react'
import ButtonPanel from '../Buttons'
import './index.css'

const Calculator = ({
	display, 
	store, 
	runningTotal,
	handleChange, 
	calculate, 
	onSubmit 
}) =>
	<form onSubmit={onSubmit}>
		Store: <input
			className="store"
			type="text"
			value={store}
			disabled
		/>
		<br/>
		Display: <input
			className="display"
			type="text"
			placeholder={display}
			value={display}
			disabled
		/>
		<br/>
		<button onClick={() => calculate()}>
			Calculate
		</button>
		<ButtonPanel handleChange={handleChange} />
	</form>

export default Calculator
