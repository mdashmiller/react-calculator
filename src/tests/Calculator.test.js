import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { spy } from 'sinon'

import Calculator from '../Calculator'
import ButtonPanel from '../Buttons'

Enzyme.configure({ adapter: new Adapter() })

describe('<Calculator />', () => {

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<Calculator />, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	it('has a valid snapshot', () => {
		const component = renderer.create(
			<Calculator />
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 form element', () => {
		const wrapper = shallow(<Calculator />)
		expect(wrapper.find('form').length).toBe(1)
	})

	it('renders 1 store div', () => {
		const wrapper = shallow(<Calculator />)
		expect(wrapper.find('div.store').length).toBe(1)
	})

	it('renders 1 display input', () => {
		const wrapper = shallow(<Calculator />)
		expect(wrapper.find('input.display').length).toBe(1)
	})

	it('renders 1 <ButtonPanel /> component', () => {
		const wrapper = shallow(<Calculator />)
		expect(wrapper.find(ButtonPanel).length).toBe(1)
	})

	//it('handles submit events', () => {})

	//it('handles change events', () => {})
	
	//it('handles click events', () => {})

})
