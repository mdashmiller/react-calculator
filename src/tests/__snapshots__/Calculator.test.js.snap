// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calculator /> has a valid snapshot 1`] = `
<div>
  <h1>
    REACTulator
  </h1>
  <form>
    <div
      className="store"
    >
      <span />
    </div>
    <input
      className="display"
      disabled={true}
      type="text"
    />
    <br />
    <div
      className="wrapper"
    >
      <div
        className="top-row"
      >
        <input
          className="button func"
          type="button"
          value="ce"
        />
        <input
          className="button func"
          type="button"
          value="c"
        />
        <input
          className="button mod"
          type="button"
          value="+/-"
        />
      </div>
      <div
        className="button-row"
      >
        <input
          className="button number"
          type="button"
          value="7"
        />
        <input
          className="button number"
          type="button"
          value="8"
        />
        <input
          className="button number"
          type="button"
          value="9"
        />
        <input
          className="button op"
          type="button"
          value="/"
        />
      </div>
      <div
        className="button-row"
      >
        <input
          className="button number"
          type="button"
          value="4"
        />
        <input
          className="button number"
          type="button"
          value="5"
        />
        <input
          className="button number"
          type="button"
          value="6"
        />
        <input
          className="button op"
          type="button"
          value="x"
        />
      </div>
      <div
        className="button-row"
      >
        <input
          className="button number"
          type="button"
          value="1"
        />
        <input
          className="button number"
          type="button"
          value="2"
        />
        <input
          className="button number"
          type="button"
          value="3"
        />
        <input
          className="button op"
          type="button"
          value="-"
        />
      </div>
      <div
        className="button-row"
      >
        <input
          className="button number"
          type="button"
          value="0"
        />
        <input
          className="button mod"
          type="button"
          value="."
        />
        <input
          className="button func"
          id="calc"
          onClick={[Function]}
          type="button"
          value="="
        />
        <input
          className="button op"
          type="button"
          value="+"
        />
      </div>
    </div>
  </form>
</div>
`;
